docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+non-compliant licensing schemes or bad packaging practices. Looking forward, a set of guidelines
+in the plugin development documentation should be added.
+
+### Milestone 2: Building conda-based installers for napari
+
+Implementing the `constructor` workflow on napari was mainly done in a single, long-lived PR, that
+has seen a couple of minor updates in the recent releases:
+
+* Main PR: https://github.com/napari/napari/pull/3378, superseded by
+  https://github.com/napari/napari/pull/3555
+* Other PRs:
+    * https://github.com/napari/napari/pull/3462
+    * https://github.com/napari/napari/pull/4185
+    * https://github.com/napari/napari/pull/4210
+    * https://github.com/napari/napari/pull/4221
+    * https://github.com/napari/napari/pull/4307
+    * https://github.com/napari/napari/pull/4309
+    * https://github.com/napari/napari/pull/4387
+    * https://github.com/napari/napari/pull/4444
+    * https://github.com/napari/napari/pull/4447
+
+
+Adding the missing pieces to `constructor` involves changes in four different projects:
+
+* `conda/constructor` itself, the tool that builds the installer
+* `conda/menuinst`, responsible of creating the shortcuts
+* `conda/conda`, which relies on `menuinst` to delegate the shortcut creation, but this code path
+  was only enabled on Windows so far
+* `conda-forge/conda-standalone-feedstock`, which freezes a `conda` install along with their
+  dependencies and adds a thin layer on top to communicate with the `constructor` installer during
+  the installation process.
+
+The full list of PRs is available on
+[this issue comment](https://github.com/napari/napari/issues/4248#issuecomment-1066574137).
+A complete list of features can be found in the packaging documentation [^napari-packaging-docs].
+
+### Milestone 3: Adding support for conda packages in the plugin manager
+
+Support for conda/mamba handling of plugin installations was implemented in a base
+PR and then extended with different PRs:
+
+* Initial PR: https://github.com/napari/napari/pull/2943
+* Other PRs that improved and extended the funcionality:
+    * https://github.com/napari/napari/pull/4074
+    * https://github.com/napari/napari/pull/3369
+    * https://github.com/napari/napari/pull/3288
+
+Adding the necessary functionality to the plugin dialog, including:
+
+* Adapting the current `Installer` class to support not only `pip` but also `conda/mamba` in a
+  single code base.
+* Adding support for `install`, `uninstall`, and `update` commands.
+* Populating the listing with data obtained from the napari hub API [^napari-hub-api].
+* Updating the listing to filter and mark plugins based on their availability on conda-forge.
+
+Some UX/UI improvements that are not part of this milestone but could be implemented in future
+releases include:
+
+* Making the plugin dialog a side panel / side dockwidget (similar to VS Code
+  [^vscode-extensions-ui])
```

Comment:
fwiw, I don't love this UX personally, as it's hard to get details about the plugin in such a small space. I've multiple times installed a different VSCode plugin to the one I actually wanted using this method.

Reply:
I don't hate it conceptually.  But certainly acknowledge that the specific implementation is important

Reply:
This would benefit from some design explorations, prototypes, and user testing.

Reply:
Added some more conditional tenses here to clarify these are only ideas / suggestions, not requirements per se.